### DISCUSSION TITLE
ARM Linux binary support in releases?

### DISCUSSION THREAD

**Discussion by idcrook** [espanso/espanso#1176] [Feature requests and Ideas]
Thanks for this great utility application!

With an ARM Linux binary in the release, it would be possible to install more easily on a Raspberry Pi and other arm systems. On a Raspberry Pi 4 especially, the Raspberry Pi platform is adequate for daily desktop use, with Ubuntu and Raspbian providing excellent support.

Perhaps `arm.deb` and `arm64.deb`? I think cargo might make this not too complicated? but I don't have direct expertise in rust packaging.

**Comment by federico-terzi** [comment]
Hey,
Thank you for your feedback! This might be useful to many people, so I'll add it to the roadmap :)
By the way, if you can manage to use Snap on  Raspbian then I think the ARM build is already available (as it is built automatically on the snapcraft build servers).

Cheers :)

**Comment by kahironimashte** [comment]
On a Raspbery 3 B  with Raspbian 10 it is unfortunately not working via snap installation.  

`bash: /usr/local/bin/espanso: cannot execute binary file: Exec format error`

**Comment by federico-terzi** [comment]
@kahironimashte I'm currently working on a refactor that will hopefully make compiling espanso much easier. At that point, your best bet would be to compile it directly on the Raspberry PI :)

**Comment by kahironimashte** [comment]
Thanks for the effort! I really enjoy the software so far 🥇

**Comment by chrisbeardy** [comment]
I would also love a pi install file to use espanso on my pi alongside my other computers. I am not familiar with rust but would be ok to compile myself at least for starters on a pi. It would great if in the documentation, or readme or separate CONTRIBUTING.md file instructions to compile could be given?

**Comment by federico-terzi** [comment]
Hey @chrisbeardy,
I'll make sure to post some detailed compile information once the big refactor is ready! I'd be happy to add support for Raspberry PI's if the community is willing to help :)

**Comment by Vistaus** [comment]
I support this. I love Espanso on my (Linux) laptop and I would love to use it on Ubuntu Touch (ARM) as well! :)

I'm fine with a manual installation instead of a DEB file if needed.

**Comment by Vistaus** [comment]
> I support this. I love Espanso on my (Linux) laptop and I would love to use it on Ubuntu Touch (ARM) as well! :)
> 
> I'm fine with a manual installation instead of a DEB file if needed.

I'd be happy to offer a bounty of €20 to make it work on ARM so that I can use it on Ubuntu Touch.

**Comment by federico-terzi** [comment]
@Vistaus Thank you for the support! Once we set up a proper CI pipeline for v2, we should be able to offer support for ARM as well, though I don't have any ETA yet :)

> I can use it on Ubuntu Touch

I've never used Ubuntu Touch myself, do you know what desktop environment does it use? (X11 based vs Wayland)

**Comment by Vistaus** [comment]
No problem - looking forward to the support! :-)

Most Ubuntu Touch ports use Mir, but will be ported to Wayland. Some ports already use Wayland, like on the Pinephone. Espanso dev-1.x already works on Wayland, so all it takes is ARM support in order for it to work on Ubuntu Touch on the Pinephone (and future other ports).

**Comment by federico-terzi** [comment]
@Vistaus I see, thank you! If that's the case, we'll definitely need #594 first, as before version 2 there was no support for wayland :)
